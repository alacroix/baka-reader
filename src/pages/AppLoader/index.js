import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';

import appStyle from 'bakareader/src/appStyle';
import { Page } from 'bakareader/src/components';
import {
  createBooksDirectory,
  formatBook,
  getUserImportedBooks,
} from 'bakareader/src/services/FileSystem';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    fontSize: appStyle.font.size.title,
    fontWeight: 'bold',
    textAlign: 'center',
  },
  statusContainer: {
    position: 'absolute',
    width: '100%',
    bottom: '10%',
  },
  status: {
    fontSize: appStyle.font.size.default,
    textAlign: 'center',
  },
});

type PropsType = {
  navigation: any,
};

class AppLoader extends Component {
  static navigationOptions = {
    header: null,
  };

  componentWillMount() {
    createBooksDirectory();
    getUserImportedBooks()
      .then((rawBooks) => {
        this.importRawBooks(rawBooks);
      });
  }

  importRawBooks(rawBooks) {
    rawBooks.forEach((book) => {
      formatBook(book)
        .then(() => { console.log('done in index'); })
        .catch((err) => console.log(err));
    });
  }

  props: PropsType;

  render() {
    return (
      <Page noMargin>
        <View style={styles.container}>
          <Text style={styles.title}>Baka Reader</Text>
        </View>
        <View style={styles.statusContainer}>
          <Text style={styles.status}>Initialisation...</Text>
        </View>
      </Page>
    );
  }
}

export default AppLoader;
