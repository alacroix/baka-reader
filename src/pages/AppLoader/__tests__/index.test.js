import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import AppLoader from '../index';

jest.mock('../../../services/FileSystem');

describe('AppLoader', () => {
  it('should render itself', () => {
    const tree = renderer.create(
      <AppLoader />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
