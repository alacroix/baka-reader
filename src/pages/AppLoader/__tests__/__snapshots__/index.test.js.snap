// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLoader should render itself 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
      Object {
        "backgroundColor": undefined,
        "paddingHorizontal": 0,
        "paddingTop": 16,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 36,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Baka Reader
    </Text>
  </View>
  <View
    style={
      Object {
        "bottom": "10%",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 13,
          "textAlign": "center",
        }
      }
    >
      Initialisation...
    </Text>
  </View>
</View>
`;
